### ISSUE TITLE
Assimp build problem on Windows

### ISSUE THREAD

**Issue by Alex533704** [assimp/assimp#3488]
Hello i have issues compiling Assimp from source  on  windows.
I have Win10.

I want to compile Assimp 1.0.4 from source.

I use MSYS to compile Assimp.

I successfully configured Assimp with cmake.exe  then i  started to build  Assimp   using make.exe  but during the compilation some error messages  appeared and the compilation failed.

error message:

`
/usr/lib/gcc/x86_64-pc-msys/9.3.0/../../../../x86_64-pc-msys/bin/as: CMakeFiles/assimp.dir/IFCReaderGen1.cpp.o: too many sections (84889)
/tmp/cc5zllrN.s: Assembler messages:
/tmp/cc5zllrN.s: Fatal error: can't write 171 bytes to section .text of CMakeFiles/assimp.dir/IFCReaderGen1.cpp.o: 'file too big'
/usr/lib/gcc/x86_64-pc-msys/9.3.0/../../../../x86_64-pc-msys/bin/as: CMakeFiles/assimp.dir/IFCReaderGen1.cpp.o: too many sections (84889)
/tmp/cc5zllrN.s: Fatal error: can't close CMakeFiles/assimp.dir/IFCReaderGen1.cpp.o: file too big
make[2]: *** [code/CMakeFiles/assimp.dir/build.make:1591: code/CMakeFiles/assimp.dir/IFCReaderGen1.cpp.o]  Error 1
make[1]: *** [CMakeFiles/Makefile2:326: code/CMakeFiles/assimp.dir/all]  Error  2
make: *** [Makefile:150: all]  Error 2  `
 
Please help me.

**Comment by kimkulling** [comment]
You need to use a newer version of the g++, the older ones have issues with the IFC-stuff. Try to use the latest greatest,

**Comment by Alex533704** [comment]
Hello, thank for your help.

I installed in MSYS the newest GCC compiler: 10.2.0
I tried to build Assimp but during compilation (using make.exe)    a list of new errors appeared and the compilation failed.

` 

`**TEST**`

In file included from /c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/fast_atof.h:26,
                 from /c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/irrXML.cpp:12:
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h: In function 'int Assimp::ASSIMP_stricmp(const char*, const char*)':
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h:143:14: **ERROR:** '::strcasecmp' has not been declared; did you mean 'strncmp'?
  143 |     return ::strcasecmp(s1,s2);
      |              ^~~~~~~~~~
      |              strncmp
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h: In function 'int Assimp::ASSIMP_strincmp(const char*, const char*, unsigned int)':
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h:192:14: **ERROR:** '::strncasecmp' has not been declared; did you mean 'strncmp'?
  192 |     return ::strncasecmp(s1,s2, n);
      |              ^~~~~~~~~~~
      |              strncmp
make[2]: *** [contrib/irrXML/CMakeFiles/IrrXML.dir/build.make:83: contrib/irrXML/CMakeFiles/IrrXML.dir/irrXML.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:298: contrib/irrXML/CMakeFiles/IrrXML.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
`     

Help me please

**Comment by Alex533704** [comment]
Hello, thank for your help.

I installed in MSYS the newest GCC compiler: 10.2.0
I tried to build Assimp but during compilation (using make.exe)    a list of new errors appeared and the compilation failed.


```
In file included from /c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/fast_atof.h:26,
                 from /c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/irrXML.cpp:12:
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h: In function 'int Assimp::ASSIMP_stricmp(const char*, const char*)':
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h:143:14: ERROR: '::strcasecmp' has not been declared; did you mean 'strncmp'?
  143 |     return ::strcasecmp(s1,s2);
      |              ^~~~~~~~~~
      |              strncmp
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h: In function 'int Assimp::ASSIMP_strincmp(const char*, const char*, unsigned int)':
/c/Users/MY_NAME/Downloads/assimp-4.0.1/assimp-4.0.1/contrib/irrXML/./../../code/StringComparison.h:192:14: ERROR:
'::strncasecmp' has not been declared; did you mean 'strncmp'?
  192 |     return ::strncasecmp(s1,s2, n);
      |              ^~~~~~~~~~~
      |              strncmp
make[2]: *** [contrib/irrXML/CMakeFiles/IrrXML.dir/build.make:83: contrib/irrXML/CMakeFiles/IrrXML.dir/irrXML.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:298: contrib/irrXML/CMakeFiles/IrrXML.dir/all] Error 2
make: *** [Makefile:150: all] Error 2

```
Help me please

**Comment by kimkulling** [comment]
Ah, you are using version 4.0.1. The issue is fixed on 5.0.1 or on the latest master.

**Comment by Alex533704** [comment]
Hello, you told me to compile Assimp 5.0.1 instead of 4.0.1 , so i tried to compile Assimp 5.0.1 but during compilation errors appeared and the compilation failed.
I'm using MSYS with GCC 10.2.0.

Error message:
```
/usr/bin/windres: /c/Users/MY_NAME/Downloads/assimp-5.0.1/contrib/zlib/win32/zlib1.rc:7: syntax error
make[2]: *** [contrib/zlib/CMakeFiles/zlib.dir/build.make:278: contrib/zlib/CMakeFiles/zlib.dir/win32/zlib1.obj] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:272: contrib/zlib/CMakeFiles/zlib.dir/all] Fehler 2
make: *** [Makefile:150: all] Fehler 2
```
